spmd/tensor/ops/matrix_ops.py

```diff
@@ -58,16 +58,28 @@ def bmm_rules(op_schema: OpSchema) -> OutputSharding:
 @register_prop_rule("aten.baddbmm.default")
 def baddbmm_rules(op_schema: OpSchema) -> OutputSharding:
     input_spec, mat1_spec, mat2_spec = op_schema.args_spec
-    bmm_output_spec = bmm_rules(
-        OpSchema((mat1_spec, mat2_spec), {})
-    ).output_spec
-    if bmm_output_spec is None:
-        # TODO: add suggestion
-        return OutputSharding(None)
+    bmm_output_sharding = bmm_rules(OpSchema((mat1_spec, mat2_spec), {}))
+    if bmm_output_sharding.output_spec is None:
+        # TODO: add more suggestions
+        if bmm_output_sharding.schema_suggestions is not None:
+            input_suggestion = bmm_output_sharding.schema_suggestions[0]
+            # reconstruct baddbmm output sharding
+            bmm_output_sharding.schema_suggestions[0] = OpSchema(
+                args_schema=(
+                    input_spec,
+                    input_suggestion.args_schema[0],
+                    input_suggestion.args_schema[1],
+                ),
+                kwargs_schema=op_schema.kwargs_schema,
+            )
+            return bmm_output_sharding
```

Comment:
N00b question, looks like we are doing sharding propagation in https://github.com/pytorch/tau/blob/main/spmd/tensor/dispatch.py#L155. And this "output_sharding" is the input for the sharding propagation. Then why do we name it with "output"?

Reply:
Output_Sharding refers to the output sharding after a op propagation. in dispatch it's also the output of a shard_prop_func? It's just that in case output sharding is not successful we will use the info to reshard inputs